SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialStrategySystem.cpp

```diff
@@ -1,58 +1,306 @@
+					AuthorityIntentV2& AuthIntent = AuthorityIntentView.FindChecked(Delta.EntityId);
+					if (AuthEpoch == AuthIntent.AssignmentCounter)
+					{
+						EntitiesACKMigration.Add(Delta.EntityId);
+					}
+				}
+				if (Update.ComponentId == SpatialConstants::NET_OWNING_CLIENT_WORKER_COMPONENT_ID)
+				{
+					Schema_Object* NetOwnerObject = Schema_GetComponentUpdateFields(Update.Update);
+					NetOwningClientWorker& NetOwner = NetOwningClientView.FindChecked(Delta.EntityId);
+					NetOwner.ApplyComponentUpdate(Update.Update);
+					EntitiesClientChanged.Add(Delta.EntityId);
+				}
+			}
 		}
 		break;
 		case EntityDelta::ADD:
 		{
-			// TODO
+			const SpatialGDK::EntityViewElement& Element = LBView.GetView().FindChecked(Delta.EntityId);
+			{
+				const SpatialGDK::ComponentData* IntentData = Element.Components.FindByPredicate(
+					SpatialGDK::ComponentIdEquality{ SpatialConstants::AUTHORITY_INTENTV2_COMPONENT_ID });
+				check(IntentData);
+				AuthorityIntentView.Add(Delta.EntityId, AuthorityIntentV2(IntentData->GetUnderlying()));
+			}
+
+			{
+				const SpatialGDK::ComponentData* DelegationData = Element.Components.FindByPredicate(
+					SpatialGDK::ComponentIdEquality{ SpatialConstants::AUTHORITY_DELEGATION_COMPONENT_ID });
+				check(DelegationData);
+				AuthorityDelegationView.Add(Delta.EntityId, AuthorityDelegation(DelegationData->GetUnderlying()));
+			}
+
+			{
+				const SpatialGDK::ComponentData* NetOwnerData = Element.Components.FindByPredicate(
+					SpatialGDK::ComponentIdEquality{ SpatialConstants::NET_OWNING_CLIENT_WORKER_COMPONENT_ID });
+				if (ensureAlwaysMsgf(NetOwnerData, TEXT("Entity %llu missing net owner data"), Delta.EntityId))
+				{
+					auto& NetOwner = NetOwningClientView.Add(Delta.EntityId, NetOwningClientWorker(NetOwnerData->GetUnderlying()));
+					if (NetOwner.ClientPartitionId.IsSet())
+					{
+						EntitiesClientChanged.Add(Delta.EntityId);
+					}
+				}
+			}
+
+			for (auto& Storage : DataStorages)
+			{
+				Storage->OnAdded(Delta.EntityId, Element);
+			}
 		}
 		break;
 		case EntityDelta::REMOVE:
+		{
+			for (auto& Storage : DataStorages)
+			{
+				Storage->OnRemoved(Delta.EntityId);
+			}
+			AuthorityIntentView.Remove(Delta.EntityId);
+			PendingMigrations.Remove(Delta.EntityId);
+		}
+		break;
 		case EntityDelta::TEMPORARILY_REMOVED:
 		{
-			// TODO
+			for (auto& Storage : DataStorages)
+			{
+				Storage->OnRemoved(Delta.EntityId);
+			}
+			const SpatialGDK::EntityViewElement& Element = LBView.GetView().FindChecked(Delta.EntityId);
+			for (auto& Storage : DataStorages)
+			{
+				Storage->OnAdded(Delta.EntityId, Element);
+			}
 		}
 		break;
 		default:
 			break;
 		}
 	}
+
+	Strategy->Advance(Connection);
 }
 
-void SpatialStrategySystem::Flush(SpatialOSWorkerInterface* Connection)
+void FSpatialStrategySystem::Flush(ISpatialOSWorker& Connection)
 {
-	// TODO
+	PartitionsMgr->Flush(Connection);
```

Comment:
There's a lot going on here, could we get an overview of the responsibilities here, and the ordering it's happening in.

Reply:
Added a couple of comments. Order is a bit arbitrary outside of fanning out the actor's updates.